### PR TITLE
feat(sql): add `show parameters` SQL

### PR DESCRIPTION
**Pull request by bluestreak01**
Added 
```sql
show parameters
```

image

The view includes _effective_ server configuration that includes:

- property path
- env variable name
- effective value
- value source [default, env, conf]
- senstitive flag (for passwords, values are not printed)
- dynamic (future proofing, for properties that can be updated without restart)

Additionally **all** `show` SQLs are full members of SQL dialect, in that output of `show` can be filtered, ordered, groupped etc. For example these are valid SQLs:

```sql
(show parameters) where value ilike '%C:%';
(show parameters) where property_path not in ('cairo.root', 'cairo.sql.backup.root') order by 1;
(show parameters) where value_source = 'env';

```

Ent: https://github.com/questdb/questdb-enterprise/pull/272

### CODE REVIEW

File: core/src/main/java/io/questdb/griffin/SqlKeywords.java [hunk 1]
```diff
@@ -1980,6 +1980,25 @@ public static boolean isSemicolon(CharSequence token) {
         return Chars.equals(token, ';');
     }
 
+    public static boolean isServerConfKeyword(CharSequence tok) {
+        if (tok.length() != 11) {
+            return false;
+        }
+
+        int i = 0;
+        return (tok.charAt(i++) | 32) == 's'
```
Comment (ideoma) [inline core/src/main/java/io/questdb/griffin/SqlKeywords.java:1989]: nit: other dbs have syntax as 

```SQL
SHOW ALL -- Postgres
SHOW VARIABLES -- MySQL, Snowflake, Single Store
SHOW PARAMETERS -- Oracle
exec sp_configure -- MSSQL
```

**Review by ideoma — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 439 / 457 (96.06%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfiguration.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/griffin/SqlParserCallback.java|2|4|50.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/ShowServerConfCursorFactory.java|46|58|79.31%|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|66|67|98.51%|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|139|140|99.29%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|45|45|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/table/AllTablesFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/ExpressionParser.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/model/QueryModel.java|78|78|100.00%|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompilerImpl.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlKeywords.java|21|21|100.00%|
|:large_blue_circle:|io/questdb/ConfigPropertyValueImpl.java|8|8|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 424 / 442 (95.93%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfiguration.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/griffin/SqlParserCallback.java|2|4|50.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/ShowServerConfCursorFactory.java|44|56|78.57%|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|66|67|98.51%|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|139|140|99.29%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|45|45|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/table/AllTablesFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/ExpressionParser.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/model/QueryModel.java|78|78|100.00%|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompilerImpl.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlKeywords.java|21|21|100.00%|
|:large_blue_circle:|io/questdb/ConfigPropertyValueImpl.java|8|8|100.00%|

File: core/src/main/java/io/questdb/griffin/SqlKeywords.java (hunk 1 above)
Comment (bluestreak01) [reply core/src/main/java/io/questdb/griffin/SqlKeywords.java]: done, it is called `show parameters` now

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 424 / 442 (95.93%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfiguration.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/griffin/SqlParserCallback.java|2|4|50.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/ShowParametersCursorFactory.java|44|56|78.57%|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|66|67|98.51%|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|140|141|99.29%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|45|45|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/table/AllTablesFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/ExpressionParser.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/model/QueryModel.java|78|78|100.00%|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompilerImpl.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlKeywords.java|20|20|100.00%|
|:large_blue_circle:|io/questdb/ConfigPropertyValueImpl.java|8|8|100.00%|

**Pull request merged**